docs/ARCHITECTURE.md

```diff
@@ -0,0 +1,77 @@
+
+### Node Services
+The node layer comprises all facilities in Habitat that are shared between communities. These include:
+
+* Reverse proxy
+* Distributed file system
+* Process manager
+* Data sources
+
+#### Reverse Proxy
+An important part of the design philosophy of Habitat is to open as few ports as necessary. Since a Habitat instance could potentially run dozens of applications at at time, a reverse proxy is used to map requests to the proper application based on the path expressed in the URL.
+
+The reverse proxy is served on port 2041. In addition, the reverse proxy is available via `libp2p` transport on port 2042. The `libp2p` reverse proxy just redirects to the first.
+
+There are two main types of reverse proxy rules. Redirect rules redirect traffic to another server running on the nodee. In the future, this should also be able to redirect traffic to another node as well. The second is file server rules, which allow for a directory of files to be served, most likely for web content. 
+
+In the future, the reverse proxy might integrate more tightly with the data proxying facilities in Habitat.
+
+#### Distributed File System
+Habitat will eventually come packaged with a distributed file system, resembling IPFS. The most important featured of this system is content addressing, which creates a number of benefits for running the filesystem in a distributed environment.
+
+Until the distributed file system can be implemented, an instance of IPFS is run alongside Habitat. 
+
+#### Process Manager
+The process manager allows for applications to be started and stopped on a Habitat node. When a process is started, a child process is created and added to the same process group as the Habitat server. If the Habitat server is killed, all processes will also be destroyed.
+
+In the future, we should consider whether this functionality should mostly be handled by systemd (and equivalent on other OSes).
+
+#### Data Sources
+To better facilitate data sharing across nodes, Habitat communities can publish "data sources" that conform to a specific schema specified in a schema registry. Depending on permissions, these data sources can be publicly available, allowing for the creation of aggregating applications, which can range from social networks to feed readers. 
+
+Since the ultimate source of data might not be contained by a node fielding a request, the `data proxy` can direct requests to the node that is able to serve teh correct data.
```

Comment:
typo: teh -> the

maybe add 1 sentence about permissions even though its not implemented. or meh i could add it later